src/Microsoft.VisualStudio.SolutionPersistence/Model/SolutionArgumentException.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.VisualStudio.SolutionPersistence.Model;
+
+/// <summary>
+/// Reasons the SolutionArgumentException was raised.
+/// </summary>
+public enum SolutionArgumentExceptionType
+{
+    CannotMoveFolderToChildFolder,
```

Comment:
It might be good to leave a "None" value set to 0 here in case it is needed. Or just start the enum values a 1 so it could be added later.

Reply:
Good point! Will name it `Undefined` as `None` might give the impression to readers that there is no error which I see being potentially confusing :)